src/Middleware/WebSockets/src/WebSocketMiddleware.cs

```diff
@@ -249,28 +284,42 @@ public static bool CheckSupportedWebSocketRequest(string method, IHeaderDictiona
             {
                 return false;
             }
-            foundHeader = false;
 
-            values = requestHeaders.GetCommaSeparatedValues(HeaderNames.Upgrade);
+            return HandshakeHelpers.IsRequestKeyValid(requestHeaders.SecWebSocketKey.ToString());
```

Comment:
https://datatracker.ietf.org/doc/html/rfc8441#section-5
> Implementations using this extended CONNECT to bootstrap WebSockets
   do not do the processing of the Sec-WebSocket-Key and Sec-WebSocket-
   Accept header fields of [[RFC6455](https://datatracker.ietf.org/doc/html/rfc6455)] as that functionality has been
   superseded by the :protocol pseudo-header field.

We can avoid this work for H2

Reply:
CheckSupportedWebSocketRequestH2 covers HTTP/2